File: 350.py
# 350. 两个数组的交集 II
# https://leetcode-cn.com/problems/intersection-of-two-arrays-ii/
from typing import List
from collections import Counter
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        counts = Counter(nums1)
        res = []

        for num in nums2:
            if counts[num] > 0:
                res += num,
                counts[num] -= 1

        return re

s = Solution()
print(s.intersect([1,2,2,1],[2,2]))

File: c++_leetcode/350.cpp
// 350. 两个数组的交集 II
// https://leetcode-cn.com/problems/intersection-of-two-arrays-ii/
using namespace std;
#include <vector>
#include <unordered_map>
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> ans;
        unordered_map<int,int> m;
        for(auto i : nums1){
            m[i]++;
        }
        for(int a : nums2){
            if(m.count(a)){
                ans.push_back(a);
                m[a]--;
                if(m[a] == 0){
                    m.erase(a);
                }
            }
        }
        return ans;
    }
};